### ISSUE TITLE
Support logits along with raw probabilities in Bernoulli and Categorical

### ISSUE THREAD

**Issue by neerajprad** [pyro-ppl/pyro#415]
From our experience with the SS-VAE tutorial, a lot of issues with numerical stability that potentially arise in discrete distributions could be better handled if the distributions were to take in log probabilities (or log odds) directly. The tensorflow library is a good place to look at for relatively robust implementation of Bernoulli and Categorical.
